File: python/student_record.py
students_records = {
	"Courses": {  "Math", "Physics", "Computer Science", "Biology", "Chemistry",
        "Statistics", "English", "Economics", "History", "Philosophy", "Sociology",
        "Art", "Music", "Engineering", "Law", "Medicine", "Business", "Political Science",
	"Geography", "Psychology"
}
}




def register_student() :
	global students_records
	user_name = input("Enter student username (press Ok to stop):  ").strip()
	if user_name.lower() == "ok" :
		return

	student_details = {}
	student_details["Name"] = input("Enter student name:  ").strip()
	student_details["Age"] = int(input("Enter student age:  "))


	if student_details["Age"] < 18 or student_details["Age"] > 50 :
		print("Age must be between 18 and 50")
		return


	student_course = set()
	print("Enter courses for student (press Ok to stop):  ")

	while True :
		course = input("courses:  ")
		if course.lower() == "ok" :
            		break

		if course in students_records["Courses"] :
			student_course.add(course)
		else :
			print("Course not available. Available courses: ", students_records["Courses"])


	student_details["Courses"] = student_course
	student_details["Address"] = {
		"City": input("Enter student city:  ").strip().lower(),
		"Zip_code": input("Enter zip code:  ").strip()
	}

	students_records[user_name] = student_details
	print(f"Student {user_name} added successfully!\n")




def display_student_info(user_name) :
	global students_records
	print("Student data", students_records[user_name])



def display_student_course(user_name) :
	global students_records
	print("Student courses", students_records[user_name]["Courses"])



def display_student_zip_code(user_name) :
	global students_records
	print("Student ZipCode", students_records[user_name]["Address"]["Zip_code"])



def display_student_city(user_name) :
	global students_records
	print("Student City", students_records[user_name]["Address"]["City"])



def add_to_student_course(user_name, course) :
	global students_records
	if course in students_records["Courses"] :
		students_records[user_name]["Courses"].add(course)
	else :
		print("Course not available. Available courses: ", students_records["Courses"])
	print("Updated student course", students_records[user_name]["Courses"])



def add_courses_to_general_courses(course) :
	global students_records
	if course not in students_records["Courses"] :
		students_records["Courses"].add(course)
	else :
		print("Course is already offered")
	print(students_records["Courses"])



def remove_courses_from_general_courses(course) :
	global students_records
	if course in students_records["Courses"] :
		students_records["Courses"].remove(course)
	else :
		print("Course is not offered before nah")
	print(students_records["Courses"])



def update_username(old_username, new_username) :
	global students_records
	if old_username in students_records :
		students_records[new_username] = students_records.pop(old_username)
	else :
		print(f"Username '{old_username}' not found")



def update_name(user_name, new_name) :
	global students_records
	if user_name in students_records :
		students_records[user_name]["Name"] = new_name
	else :
		print(f"username '{user_name}' not found")



def update_age(user_name, new_age) :
	global students_records
	if user_name in students_records :
		students_records[user_name]["Age"] = new_age
	else :
		print(f"username '{user_name}' not found")



def update_city(user_name, new_city) : 
	global students_records
	if user_name in students_records :
		students_records[user_name]["Address"]["City"] = new_city
	else :
		print(f"username '{user_name}' not found")



def update_zip(user_name, new_zip_code) :
	global students_records
	if user_name in students_records :
		students_records[user_name]["Address"]["Zip_code"] = new_zip_code
	else :
		print(f"username '{user_name}' not found")



def get_number_of_student_in_record() :
	global students_records
	return len([students for students in students_records if students != "Courses"])


print(" ")
print(" ")
print(" ")
def menu() :

	while True :
		print("WELCOME TO BRIGHT FUTURE UNIVERSITY")
		print("HERE WE...... WHAT DO WE DO AGAIN SELF")
		print("ANYWAY WHAT DO YOU WANNA DO??")
		print("\n==========Student Record Menu==========")
		print("1. Register new student")
		print("2. Display student info")
		print("3. Display student courses")
		print("4. Display student city")
		print("5. Display student zip code")
		print("6. Add course to student")
		print("7. Add new course to general list")
		print("8. Remove course from general list")
		print("9. Update username")
		print("10. Update student name")
		print("11. Update student age")
		print("12. Update student city")
		print("13. Update student zip code")
		print("14. Get total number of students")
		print("0. Exit")

		choice = input("Choose an option: ").strip()

		if choice == "0" :
			print("Thank you for using this app.... Goodbye!")
			break
		elif choice == "1" :
			register_student()
		elif choice == "2" :
			user_name = input("Enter username: ").strip()
			display_student_info(user_name)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "3" :
			user_name = input("Enter username: ").strip()
			display_student_course(user_name)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "4" :
			user_name = input("Enter username: ").strip()
			display_student_city(user_name)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "5" :
			user_name = input("Enter username: ").strip()
			display_student_zip_code(user_name)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "6" :
			user_name = input("Enter username: ").strip()
			course = input("Enter course to add: ").strip()
			add_to_student_course(user_name, course)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "7" :
			course = input("Enter new course to add: ").strip()
			add_courses_to_general_courses(course)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "8" :
			course = input("Enter course to remove: ").strip()
			remove_courses_from_general_courses(course)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "9" :
			old_name = input("Enter old username: ").strip()
			new_username = input("Enter new username: ").strip()
			update_username(old_name, new_username)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "10" :
			user_name = input("Enter username: ").strip()
			new_name = input("Enter new name: ").strip()
			update_name(user_name, new_name)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "11" :
			user_name = input("Enter username: ").strip()
			new_age = int(input("Enter new age: "))
			update_age(user_name, new_age)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "12" :
			user_name = input("Enter username: ").strip()
			new_city = input("Enter new city: ").strip()
			update_city(user_name, new_city)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "13" :
			user_name = input("Enter username: ").strip()
			new_zip_code = input("Enter new zip code: ").strip()
			update_zip(user_name, new_zip_code)
			print(" ")
			print(" ")
			print(" ")

		elif choice == "14" :
			print("Total students:", get_number_of_student_in_record())
			print(" ")
			print(" ")
			print(" ")

		else :
			print("You get eyes nah!!.")


menu()

File: JavaScript/StudentRecord.js
const prompt = require("prompt-sync")();

let studentsRecords = {
    "Courses": new Set([
        "Math", "Physics", "Computer Science", "Biology", "Chemistry",
        "Statistics", "English", "Economics", "History", "Philosophy", "Sociology",
        "Art", "Music", "Engineering", "Law", "Medicine", "Business", "Political Science",
	"Geography", "Psychology"
    ])
};


function registerStudent() {
    let userName = prompt("Enter student username (press Ok to stop):").trim();
    if (userName.toLowerCase() === "ok") return;

    let studentDetails = {};
    studentDetails["Name"] = prompt("Enter student name:").trim();
    studentDetails["Age"] = parseInt(prompt("Enter student age:"));

    if (studentDetails["Age"] < 18 || studentDetails["Age"] > 50) {
        console.log("Age must be between 18 and 50");
        return;
    }

    let studentCourse = new Set();
    console.log("Enter courses for student (type Ok to stop):");

    while (true) {
        let course = prompt("courses:");
        if (course.toLowerCase() === "ok") break;

        if (studentsRecords["Courses"].has(course)) {
            studentCourse.add(course);
        } else {
            console.log("Course not available. Available courses:", studentsRecords["Courses"]);
        }
    }

    studentDetails["Courses"] = studentCourse;
    studentDetails["Address"] = {
        "City": prompt("Enter student city:").trim().toLowerCase(),
        "ZipCode": prompt("Enter zip code:")
    };

    studentsRecords[userName] = studentDetails;
    console.log(`Student ${userName} added successfully!\n`);
}


function displayStudentInfo(userName) {
    console.log("Student data", studentsRecords[userName]);
}

function displayStudentCourse(userName) {
    console.log("Student courses", studentsRecords[userName]["Courses"]);
}

function displayStudentZipCode(userName) {
    console.log("Student ZipCode", studentsRecords[userName]["Address"]["ZipCode"]);
}

function displayStudentCity(userName) {
    console.log("Student City", studentsRecords[userName]["Address"]["City"]);
}


function addToStudentCourse(userName, course) {
    if (studentsRecords["Courses"].has(course)) {
        studentsRecords[userName]["Courses"].add(course);
    } else {
        console.log("Course not available. Available courses:", studentsRecords["Courses"]);
    }
    console.log("Updated student course", studentsRecords[userName]["Courses"]);
}

function addCoursesToGeneralCourses(course) {
    if (!studentsRecords["Courses"].has(course)) {
        studentsRecords["Courses"].add(course);
    } else {
        console.log("Course is already offered");
    }
    console.log(studentsRecords["Courses"]);
}

function removeCoursesFromGeneralCourses(course) {
    if (studentsRecords["Courses"].has(course)) {
        studentsRecords["Courses"].delete(course);
    } else {
        console.log("Course is not offered before nah");
    }
    console.log(studentsRecords["Courses"]);
}


function updateUsername(oldUserName, newUserName) {
    if (studentsRecords[oldUserName]) {
        studentsRecords[newUserName] = studentsRecords[oldUserName];
        delete studentsRecords[oldUserName];
    } else {
        console.log(`Username '${oldUserName}' not found`);
    }
}

function updateName(userName, newName) {
    if (studentsRecords[userName]) {
        studentsRecords[userName]["Name"] = newName;
    } else {
        console.log(`username '${userName}' not found`);
    }
}

function updateAge(userName, newAge) {
    if (studentsRecords[userName]) {
        studentsRecords[userName]["Age"] = newAge;
    } else {
        console.log(`username '${userName}' not found`);
    }
}

function updateCity(userName, newCity) {
    if (studentsRecords[userName]) {
        studentsRecords[userName]["Address"]["City"] = newCity;
    } else {
        console.log(`username '${userName}' not found`);
    }
}

function updateZip(userName, newZipCode) {
    if (studentsRecords[userName]) {
        studentsRecords[userName]["Address"]["ZipCode"] = newZipCode;
    } else {
        console.log(`username '${userName}' not found`);
    }
}


function getNumberOfStudentInRecord() {
    let count = 0;
    for (let key in studentsRecords) {
        if (key !== "Courses") count++;
    }
    return count;
}


function menu() {
    while (true) {
        console.log("\nWELCOME TO BRIGHT FUTURE UNIVERSITY");
        console.log("HERE WE...... WHAT DO WE DO AGAIN SELF");
        console.log("ANYWAY WHAT DO YOU WANNA DO??");
        console.log("\n========== Student Record Menu ==========");
        console.log("1. Register new student");
        console.log("2. Display student info");
        console.log("3. Display student courses");
        console.log("4. Display student city");
        console.log("5. Display student zip code");
        console.log("6. Add course to student");
        console.log("7. Add new course to general list");
        console.log("8. Remove course from general list");
        console.log("9. Update username");
        console.log("10. Update student name");
        console.log("11. Update student age");
        console.log("12. Update student city");
        console.log("13. Update student zip code");
        console.log("14. Get total number of students");
        console.log("0. Exit");

        let choice = prompt("Choose an option:").trim();

        if (choice === "0") {
            console.log("Thank you for using this app.... Goodbye!");
            break;
        } else if (choice === "1") {
            registerStudent();
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "2") {
            let userName = prompt("Enter username:").trim();
            displayStudentInfo(userName);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "3") {
            let userName = prompt("Enter username:").trim();
            displayStudentCourse(userName);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "4") {
            let userName = prompt("Enter username:").trim();
            displayStudentCity(userName);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "5") {
            let userName = prompt("Enter username:").trim();
            displayStudentZipCode(userName);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "6") {
            let userName = prompt("Enter username:").trim();
            let course = prompt("Enter course to add:").trim();
            addToStudentCourse(userName, course);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "7") {
            let course = prompt("Enter new course to add:").trim();
            addCoursesToGeneralCourses(course);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "8") {
            let course = prompt("Enter course to remove:").trim();
            removeCoursesFromGeneralCourses(course);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "9") {
            let oldUserName = prompt("Enter old username:").trim();
            let newUsername = prompt("Enter new username:").trim();
            updateUsername(oldUserName, newUsername);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "10") {
            let userName = prompt("Enter username:").trim();
            let newName = prompt("Enter new name:").trim();
            updateName(userName, newName);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "11") {
            let userName = prompt("Enter username:").trim();
            let newAge = parseInt(prompt("Enter new age:"));
            updateAge(userName, newAge);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "12") {
            let userName = prompt("Enter username:").trim();
            let newCity = prompt("Enter new city:").trim();
            updateCity(userName, newCity);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "13") {
            let userName = prompt("Enter username:").trim();
            let newZipCode = prompt("Enter new zip code:").trim();
            updateZip(userName, newZipCode);
	    console.log(" ");
	    console.log(" ");
        } else if (choice === "14") {
            console.log("Total students:", getNumberOfStudentInRecord());
	    console.log(" ");
	    console.log(" ");
        } else {
            console.log("You get eyes nah!!.");
        }
    }
}

menu();
